microsetta_interface/templates/new_results_page.jinja2

```diff
@@ -594,6 +594,439 @@
         select.prop("disabled", false);
     }
 
+    function drawBoxPlots(top5) {
+        let mockData = top5.map(d => {
+            const relativeAbundance = parseFloat(d.relativeAbundance);
+
+            let maxRange;
+            if (relativeAbundance > 50) {
+                maxRange = 100;
+            } else if (relativeAbundance < 50 && relativeAbundance > 5) {
+                maxRange = 50;
+            } else {
+                maxRange = 5;
+            }
+
+            const minVal = 0;
+            const maxVal = maxRange;
+            const q1 = relativeAbundance * 0.25;
+            const median = relativeAbundance * 0.5;
+            const q3 = relativeAbundance * 0.75;
+
+            return {
+                genus: d.Genus || d.genus || "Unknown",
+                relativeAbundance: relativeAbundance,
+                stats: { min: minVal, q1: q1, median: median, q3: q3, max: maxVal }
+            };
+        });
+
+        const columns = 2;
+        const panelWidth = 600;
+        const panelHeight = 240;
+        const spacingX = 40;
+        const spacingY = 70;
+
+        const totalWidth = 1280;
+        const totalBoxWidth = columns * panelWidth + (columns - 1) * spacingX;
+        const remainingWidth = totalWidth - totalBoxWidth;
+
+        const rows = Math.ceil(mockData.length / columns);
+
+        const margin = { top: 20, right: 0, bottom: 60, left: -20 };
+
+        const totalHeight = rows * panelHeight + (rows - 1) * spacingY + margin.top + margin.bottom;
+
+        d3.select("#boxplot-svg").selectAll("svg").remove();
+
+        const svg = d3.select("#boxplot-svg")
+            .append("svg")
+            .attr("width", totalWidth)
+            .attr("height", totalHeight);
+
+        const g = svg.append("g")
+            .attr("transform", `translate(0,${margin.top})`);
+
+        const textAnchor = "end";
+
+        mockData.forEach((d, i) => {
```

Comment:
Rather than appending each boxplot to a single SVG, the code should render an SVG for each boxplot, which should be laid out on the screen using responsive flexbox CSS code. Two columns on full-size devices, one column on smaller screens. Additionally, the size of the boxplots should be scalable to work with smaller screens that don't have 600 horizontal pixels to work with.

If scaling all of the constituent elements of the boxplot would not render appropriately, I'm open to the idea of building the SVG with a fixed width of 600px and displaying a scaled version of the final SVG with instructions for users to click on the image to "zoom in" and view it at full resolution on smaller screens. But I'd prefer to explore a fully responsive approach.